### ISSUE TITLE
[Bug][a11y]: Card component checkbox for selection has no accessibility name and is not keyboard accessible

### ISSUE THREAD

**Issue by majornista** [adobe/spectrum-web-components#5168]
### Code of conduct

- [x] I agree to follow this project's code of conduct.

### Impacted component(s)

Card, Grid

### Expected behavior

In a Card component, when the `toggles` boolean property is specified:
1. the checkbox that displays on hover must have an accessibility name.
2. the checkbox should be keyboard focusable

To fix these issues, remove  `tabIndex="-1"` and include the header text as a hidden accessible name for the sp-checkbox at:
https://github.com/adobe/spectrum-web-components/blob/e7d153e653da03fbceb0c5a605bc6e2f6d21ca3a/packages/card/src/Card.ts#L318-L323

```js
                          <sp-checkbox
                              class="checkbox"
                              @change=${this.handleSelectedChange}
                              ?checked=${this.selected}
                          >
                            <span class="sr-only">${this.heading}</span>
                          </sp-checkbox>
```

The `.sr-only` style, and a `--mod-checkbox-text-to-control` override to account for the fact that the text label is visually hidden, should be added to the [card.css](https://github.com/adobe/spectrum-web-components/blob/main/packages/card/src/card.css) file:

```css
.checkbox-toggle .checkbox {
    --mod-checkbox-text-to-control: 0;
}

.sr-only {
    position: absolute;
    width: 1px;
    height: 1px;
    padding: 0;
    margin: -1px;
    overflow: hidden;
    clip: rect(0, 0, 0, 0);
    white-space: nowrap;
    border: 0;
}
```
At present, this is the only way to appropriately label a `sp-checkbox`.

3. Similarly, in the storybook examples using the Card with an action button, the action button should be keyboard focusable.

Again, we need to remove `tabIndex="-1"` from the `sp-action-menu`  in Card rendering for the Storybook example:
https://github.com/adobe/spectrum-web-components/blob/e7d153e653da03fbceb0c5a605bc6e2f6d21ca3a/tools/grid/stories/grid.stories.ts#L77-L83

```js
             <sp-action-menu 
                 label="File actions" 
                 slot="actions" 
                 placement="bottom-end" 
                 quiet 
             > 
```

### Actual behavior

In a Card component, when the `toggles` boolean property is specified:
1. the checkbox that displays on hover or focus does not have an accessibility name.
2. the checkbox cannot receive focus with a keyboard
3. Similarly, in the storybook examples using the Card with an action button, the action button is not keyboard focusable.

### What browsers are you seeing the problem in?

Chrome, Firefox, Safari, Microsoft Edge

### Severity

None
